### PR TITLE
Don't generate invalid dates - same as 840

### PR DESCRIPTION
**Pull request by rflament**
I tried to finish the pull request about the date format error : https://github.com/vpulim/node-soap/pull/840 

It uses moment.js.

Most of the work had been done by mredbishop , I just made small changes to change attributes name and reduce the size of the files used in the test.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,8 @@
     "ejs": "~2.3.4",
     "finalhandler": "^0.5.0",
     "lodash": "^3.10.1",
+    "moment": "~2.13.0",
```
Comment (jsdevel) [inline package.json]: Not sure how crazy I am about adding `moment` as a dependency.  Is there a way to just pass in a string?

Comment (jsdevel) [inline package.json]: Thoughts @herom ?

Comment (rflament) [inline package.json]: Another way would be to be able to pass a function to deserialize specific fields by name or by type. Could be in options.

Then the code would look like :


if (customDeserializerExists(name)) {
value = customDeserializerExists(name)(text);
}
else {
 
// current code
if (name === 'int' || name === 'integer') {
      value = parseInt(text, 10);
    } else if (name === 'bool' || name === 'boolean') {
      value = text.toLowerCase() === 'true' || text === '1';
...
...

}

Comment (jsdevel) [inline package.json]: I think something like that would be better

Comment (rflament) [inline package.json]: I'll open another pull request.

Comment (rflament) [inline package.json]: https://github.com/vpulim/node-soap/pull/901

**Pull request closed without merging**